### PR TITLE
Bump cecil and use the new setter for ExportedType.Scope. Fixes #51805

### PR DESCRIPTION
**Pull request by spouliot**
Older Cecil (0.9) were saving the scope of the actual type. This changed
with Cecil 0.10 where the ExportedType.Scope is saved, which could lead
to invalid assemblies being saved with the current linker code (which
could not update the scope without the cecil bump)

reference:
https://bugzilla.xamarin.com/show_bug.cgi?id=51805

### CODE REVIEW

File: linker/Mono.Linker.Steps/SweepStep.cs [hunk 1]
```diff
@@ -175,6 +175,7 @@ bool ResolveAllTypeReferences (AssemblyDefinition assembly)
 					if ((td != null) && Annotations.IsMarked (td)) {
 						scope = assembly.MainModule.ImportReference (td).Scope;
 						hash.Add (td, scope);
```
Comment (erozenfeld) [inline linker/Mono.Linker.Steps/SweepStep.cs]: I don't think the line above your change is needed. The scope of the type definition doesn't need to be updated.

Comment (erozenfeld) [inline linker/Mono.Linker.Steps/SweepStep.cs]: The line above the if is also not needed.

Comment (marek-safar) [inline linker/Mono.Linker.Steps/SweepStep.cs]: @erozenfeld send separate PR with the clean up
